docs/ext/remix_code_links.py

```diff
@@ -0,0 +1,56 @@
+import base64
+import docutils  # pragma pylint: disable=import-error
+
+
+def insert_node_before(child, new_sibling):
+    assert child in child.parent.children
+
+    for position, node in enumerate(child.parent.children):
+        if node == child:
+            child.parent.insert(position, new_sibling)
+            break
+
+
+def remix_code_url(source_code):
+    # NOTE: base64 encoded data may contain +, = and / characters. Remix seems to handle them just
+    # fine without any escaping.
+    base64_encoded_source = base64.b64encode(source_code.encode('utf-8')).decode('ascii')
+    return f"https://remix.ethereum.org/?code={base64_encoded_source}"
```

Comment:
@axic
> Does remix has a version parameter still, or will it just find it out based on the pragma?

Too many things in a single thread so answering his in a separate one.

It does. The problem is that it looks like this: `version=soljson-v0.8.4+commit.c7e474f2.js`. So if we wanted to include version in the link then the extension would have fetch version list at build time. It would also have to parse the version pragma and choose some matching version (the newest one?).

Do you think it's worth the complication? In the latest docs most snippets work with the latest compiler so the default version will be fine most of the time. On the other hand in older docs that's not necessarily the case so maybe it's worth it just for that.

An alternative solution would be to add an option to code blocks that lets you set compiler version manually. We would then have to manually add it to code blocks though and update with each new version (but that could be automated that with a script that we would run on the docs after release and then commit the modified docs).